hlpdoc/src/configuration.txt

    Getting Started:

        Depending on the operating system, the name of the *GRIEF* executable
        is generally.

                o gr (console)
                o xgr (x11)

        The following command line switch are available:

>
>       GRIEF 3.2.2.25 compiled Apr 10 2024 00:40:33 (cm version 3.1)
>
>       Usage: cr [options] [+line-number] file ..
>
>       Options:
>         -a, --add                     Add file to current profile.
>         -b, --nobackup                Disable backup creation.
>         --noautosave                  Disable autosave.
>         -c, --compat                  Compatability mode (dont enable nodelay).
>         -E, --display                 Enable display on startup.
>         -D, --define var[=value]      Define an environment variable.
>         -e, --echoflags charval,virt,col,percent,time,remember,cursor
>                                       Echo line state information.
>         --echofmt <format-spec>       Echo line format definition.
>         -i, --timer N                 Set interval timer to N seconds.
>         -M, --vm #                    Set virtual memory size to # lines.
>         --mouse                       Enable mouse.
>         --nomouse                     Disable mouse support.
>         --scroll                      Enable xterm scrolling.
>         --noscroll                    Disable xterm scrolling.
>         --color [=<depth>]            Force color mode.
>         --nocolor                     Force black and white display.
>         --visbell                     Enable visable bell.
>         --light                       Light color scheme.
>         --dark                        or dark color scheme.
>         --utf8 [=[no|yes],[[no]combined,seperate],[subst|ncr|ucn|hex]]
>                                       UTF8 features.
>         --8bit [=<encoding>]          Eight bit terminal encoding, with optional encoding.
>         --7bit                        Seven bit terminal encoding.
>         --guess charset,chardet,mark,bom,magic,utf...
>                                       File encoding search specification.
>         --buftype dos|unix|mac|ansi   Default buffer-type.
>         --encoding <encoding>         Default file encoding.
>         --ucsver <x.y.z>              Unicode version; wcwidth support.
>         --nograph                     Disable use of graphic characters.
>         --nounicode                   Disable use of UNICODE graphic characters.
>         --nounderline                 Disable use of underline mode.
>         --nohilite                    Disable syntax hiliting.
>         --nocygwinkb                  Disable use of cygwin raw scancodes.
>         --nosigtrap                   Disable signal trapping (for debugging).
>         --term <termname>             Override the TERM setting.
>         --grterm <termname>           Override the GRTERM setting.
>         --grhelp <path>               Override the GRHELP setting.
>         --grprofile <user-profile>    Override the GRPROFILE setting.
>         --termcap                     Use termcap if available, otherwise terminfo.
>         --curses                      Enable/force use of curses tty driver.
>         --noinit                      Disable termcap init/deinit strings.
>         --nokeypad                    Disable termcap keypad init/deinit strings.
>         --noborders                   Disable use of borders (main windows/non-popup).
>         --notitle                     Disable use/update of console title.
>         --lazy [=lines]               Limit vt updates to # lines, delayed until stable; for slow links.
>         --jump [=lines]               Scroll jump, paging content not scolling; for slow links.
>         --visible [=lines]            Visible lines above/below cursor.
>         --escdelay value              ESC delay in milliseconds.
>         -U, --user <user-profile>     Execute user profile.
>         -m, --macro <macro-name>      Execute named macro.
>         -R, --readonly                Read-only mode.
>         --lockstrict                  Enable strict file-locking.
>         --spell                       Spell check.
>         --nospell                     Disable spell.
>         -t, --spaces                  Use spaces to fill out tabs.
>         -d, --log                     Create `grief.log` diagnostics log.
>         -f, --logflush                Flush log file as created.
>         -P, --dflags regexp,undo,prompt,memory,native,rtc,purify,terminal,vfs,refs,profile
>                                       Debugging/profiling flags, comma separated.
>         -W, --warnings                Enable warnings.
>         -s, --logstats                Report statistics on exit.
>         -r, --statsref                Generate macro reference statistics on exit.
>         -w, --readchar                Allow read_char to return -1.
>         --test [value]                test mode.
>         --restrict                    restrictive mode.
>         -?, --help                    Usage information.
>         -V, --config                  Configuration details.
>         -v, --version                 Version.
>         --authors                     Developer details.
>         --license                     License information.
>         +nn                           Goto line nn.
>

        No te!:
        Op tions can be system/configuration dependant and shall be
        disregarded if not applicable.

    X11:

        xgr accepts additional switches.

    Startup Files:

        When GRIEF starts up, it loads an initialization file, grief.cm.


Changes
-*- encoding: utf-8; -*-

	* BUILD 24, version: 3.2.3

Tue May 28 19:08:00 2024 adamy

	* configure upgrades

	* documentation/install

	* libguess-1.2

	* duktape 2.7.0 merge

	* libw32 merge

	* general maintance and bugfixes.

	* AltGr/WIN32

Sun Aug 28 20:44:56 2022 adamy

	* html/xml

		o tag matching/hiliting syntax flags

			- SYNF_HTMLTAG
			- SYNF_XMLTAG

		o void tags

Wed Aug 10 22:52:02 2022 adamy

	* syntax

		o syntax_find - syntax aware brace matching.

			- BF_SYNTAX_MATCH buffer flag, bufinfo
			- find_matching_brace

		* dfa, accept2 state processing.

	* tickets #19, #22

	* bug-fixes

		- window sizing issues
		- cmdline nesting
		- llong handling

Mon Jul 11 22:05:51 2022 adamy

	* INSTALLation documentation updated; install interface cleanup.

	* new modes and updated modes; json, cmake and protobuf.

	* keyword pattern/wild-card support; libtrie

Sat May 28 16:52:13 2022 adamy

	* hunspell-1.7.0

Fri May 27 20:46:49 2022 adamy

	* libarchive-3.6.1

	* mingw - alpha

		o mingw toolchain, makelib and libtool.
		o gcc build and warnings.

Thu Mar 24 13:36:07 2022 adamy

	* mandoc-1.14.6

Tue Mar 22 18:46:50 2022 adamy

	* BUILD 21, version: 3.2.2

	* libw32 merge.

Mon Jul 12 22:52:00 2021 adamy

	* echo-line winch handling.

	* width-character

		o wcwidth(), set_unique_version() and inq_unicode_version() primitives.
		o wxxx() primitives().
		o %S and %W printf formats.
		o buffer insertion primitives (insert() and insertf()) return width not characters.
		o read() primitive encoding aware.
		o wide-character aware cursor movements.
		o unicode description table, inline character-value.
		o libiconv() dll relative paths.
		o 32-bit internal key-codes.
		o Alt+Keycode handler (win32 - experimental).

	* utf8 filenames.

Mon Apr 12 20:50:30 2021 adamy

	* BUILD 17 release

	* duktape 2.6.0

Sat Apr  3 11:50:35 2021 adamy

	* gcc 10.x and glib/threads.h

	* mandoc cygwin compat issues.

Wed Jun  3 21:42:20 2020 adamy

	* appveyor

		o Ubuntu and win32 ci build support.

		- https://ci.appveyor.com/project/adamyg/grief-win
		- https://ci.appveyor.com/project/adamyg/grief-ubuntu

Mon May  4 20:10:25 2020 adamy

	* Macros

		o find - Current buffer search.
		o tree - Reworked buffer management.

Tue Apr 28 00:40:07 2020 adamy

	* Macro language

		o Register variable and lexical-scope support.
		o true/fale, plus NAN/INFINITY predefines.
		o Hex floating point syntax.
		o isclose() float comparision.
		o Cast operators (int and float/double).

	* Grunch

		o Allow concurrent cm and m generation.
		o Slashes converted within debug filenames.
		o Removed unneeded outer macro list closure.
		o Empty macros associated with nothing verb.
		o Unary + added.

	* Performance/profiling

		o Explicit symbol opcode.
		o Benchmark macros.
		o Extended regression tests.
		o Inbuilt execution streamlining.
		o Cache alignment.
		o Inlined LIST access primitives.
		o Explicit set_hooked() usage; DEBUG runtime checks catch semantic errors.
		o Removed unneed accumlator interactions.

	* General

		o Consume stray ESC keystrokes; default keyboard.
		o Documentation tweaks.
		o -+debug feature reporting.

	* Ticket #15 - Screen not properly restored (WIN32).

		o Explicit application of console rgb table.
		o Disable quick-edit during session, when mouse enabled.
		o Save and restore under 16-bit mode.
		o Restore console attribute on exit.
		o Reactive winch handling.

Sat Oct 26 00:49:25 2019 adamy

	* BUILD 17 release
	* added missing wc19 runtime dll

Wed Sep 25 23:20:14 2019 adamy

	* BUILD 16 release

Wed May  1 13:31:26 2019 adamy

	* mandoc 1.14.4 - gcc builds
	* configure; strndup, asprintf and vasprintf

Mon Feb  4 11:47:31 2019 adamy

	* duktape 2.3.0; alpha integration

Mon Nov 12 01:49:21 2018 adamy

	* version 3.2.1
	* mandoc 1.14.4

Thu Oct 11 01:52:20 2018 adamy

	* dynamic library linkage
	* windows 10 console
	* legacy console font/optimisation tweaks
	* libw32 merge
		- symlink/mount-point support

Mon Jan 30 15:42:45 2017 adamy

	* xz utils 5.2.3

Sat Jan 28 02:03:55 2017 adamy

	* libmagic 5.29

Mon Jan 23 21:59:18 2017 adamy

	* msvc 2008 - 2015 builds.

	* libz 1.2.11

Sun Mar 15 19:31:54 2015 adamy

	* emacs mode.

	* win32

		- trailing italic character truncation.
		- symlink() EEXIST error code.

Thu Feb 19 00:35:01 2015 adamy

	* BUILD8 released.

Mon Dec 15 02:55:34 2014 adamy

	* installer, corrected PATH 'bin' path.

	* full-screen console,

		- ega(), minimise/maximize emulation (aka <Alt-Enter>) under win32 console.

		- <Ctrl-Enter>, 'fullscreen' default mapping.

		- explain fullscreen.

	* ReadConsoleOutputA() ERROR_NOT_ENOUGH_MEMORY handling.

Mon Dec 15 02:55:34 2014 adamy

	* Corrected primitive sort order.

Fri Dec 12 18:35:53 2014 adamy

	* Corrected license within about/welcome dialogs.

Fri Dec 01 18:35:53 2014 adamy

	* Public release, BUILD7.

/*
 * Local Variables: ***
 * mode: changelog ***
 * End: ***
 */


gr/config.c
#include <edidentifier.h>
__CIDENT_RCSID(gr_config_c,"$Id: config.c,v 1.35 2022/08/10 15:44:56 cvsuser Exp $")

/* -*- mode: c; indent-width: 4; -*- */
/* $Id: config.c,v 1.35 2022/08/10 15:44:56 cvsuser Exp $
 * Machine dependent configuration variables.
 *
 *
 * This file is part of the GRIEF Editor.
 *
 * The GRIEF Editor is free software: you can redistribute it
 * and/or modify it under the terms of the GRIEF Editor License.
 *
 * The GRIEF Editor is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * License for more details.
 * ==end==
 */

#include "config.h"

#include <editor.h>
#include <edpaths.h>
#include <edalt.h>

/*
 *  Default environment
 */
#include <limits.h>                             /* confirm compiler environment */
#if !defined(__MAKEDEPEND__)
#if (CHAR_MIN == 0)
#error Character value is unsigned
#endif
#endif

const char *x_grfile        = "GRFILE=newfile";
const char *x_grflags       = "GRFLAGS=-i60";

#if defined(_VMS)
const char *x_machtype      = "VMS";
#elif defined(__OS2__)
const char *x_machtype      = "OS/2";
#elif defined(_WIN32) || defined(WIN32)
#if defined(__MINGW32__)
const char *x_machtype      = "Mingw32";
#else
const char *x_machtype      = "Win32";          /* 98,NT,XP etc */
#endif
#elif defined(DOSISH)
const char *x_machtype      = "DOS";
#elif defined(__APPLE__)
const char *x_machtype      = "MACOSX";
#else
const char *x_machtype      = "UNIX";           /* others including Cygwin */
#endif

#if defined(_VMS)
const char *x_grtermcap     = "grief_library:[etc]termcap.dat";
const char *x_grhelp        = "grief_library:[help]";
const char *x_grpath        = "grief_library:[macros]";
#else
const char *x_grtermcap     = _PATH_TERMCAP;
const char *x_grhelp        = _PATH_GRIEF_HELP;
#if defined(DOSISH)
const char *x_grpath        = _PATH_GRIEF_MACROS ";" _PATH_GRIEF_SOURCE;
#else
const char *x_grpath        = _PATH_GRIEF_MACROS ":" _PATH_GRIEF_SOURCE;
#endif
#endif

#if defined(__MSDOS__)
#if defined(__MINGW32__)
const char *x_default_term  = "TERM=mingw32";
#elif defined(_WIN32) || defined(WIN32)
const char *x_default_term  = "TERM=win32";
#else
const char *x_default_term  = "TERM=dos";
#endif
#elif defined(__OS2__)
const char *x_default_term  = "TERM=os2";
#elif defined(__CYGWIN__)
const char *x_default_term  = "TERM=cygwin";
#elif defined(linux)
const char *x_default_term  = "TERM=linux";
#elif defined(_AIX)
const char *x_default_term  = "TERM=aixterm";
#else
const char *x_default_term  = "TERM=ansi";
#endif


/*
 *  Default colours
 */
const char *                x_col_table_dark[] = {
    "background="               "NONE",
    "normal="                   "NONE",
    "select="                   "CYAN",
    "message="                  "GREEN",
    "error="                    "RED",
    "hilite="                   "WHITE,DARK-BLUE",
    "standout="                 "BLUE",
    "whitespace="               "WHITE,RED",

    "dialog_focus="             "MAGENTA",
    "dialog_hilite="            "LIGHT-WHITE,DARK-BLUE",
    "dialog_but_normal="        "GREY",
    "dialog_but_focus="         "LIGHT-WHITE",
    "dialog_but_key_normal="    "DARK-RED:underline",
    "dialog_but_key_focus="     "RED:underline",
    "dialog_edit_greyed="       "GREY,BLACK",
    "dialog_edit_normal="       "BLACK,GREY",
    "dialog_edit_focus="        "BLACK,WHITE",

    "string="                   "GREEN",
    "operator="                 "BLUE",
    "number="                   "GREEN",
    "comment="                  "CYAN",
    "delimiter="                "DARK-GREY",
    "preprocessor="             "MAGENTA",
    "preprocessor_keyword="     "DARK-MAGENTA",
    "keyword="                  "YELLOW",

    NULL
    };


const char *                x_col_table_light[] = {
    "background="               "NONE",
    "normal="                   "NONE",
    "select="                   "DARK-BLUE",
    "message="                  "BLACK",
    "error="                    "DARK-RED",
    "hilite="                   "WHITE,DARK-BLUE",
    "standout="                 "DARK-BLUE",

    "dialog_focus="             "DARK-MAGENTA",
    "dialog_hilite="            "WHITE,DARK-BLUE",
    "dialog_but_normal="        "BLACK",
    "dialog_but_focus="         "WHITE,BLACK",
    "dialog_but_key_normal="    "RED:underline",
    "dialog_but_key_focus="     "RED,BLACK:underline",
    "dialog_edit_greyed="       "GREY,BLACK",
    "dialog_edit_normal="       "LIGHT-WHITE,BLACK",
    "dialog_edit_focus="        "CYAN,BLACK",

    "string="                   "DARK-RED",
    "operator="                 "DARK-BLUE",
    "number="                   "DARK-RED",
    "comment="                  "DARK-BLUE",
    "delimiter="                "DARK-BLUE",
    "preprocessor="             "DARK-MAGENTA",
    "preprocessor_keyword="     "DARK-RED",
    "keyword="                  "DARK-GREEN",

    NULL
    };


/*
 *  Default borderless window colours.
 */
const char *                x_col_windows[] = {
    "window1="                  "YELLOW,BLACK",
    "window2="                  "LIGHT-WHITE,BLUE",
    "window3="                  "BLACK,GREEN",
    "window4="                  "BLUE,CYAN",
    "window5="                  "YELLOW,RED",
    "window6="                  "BLUE,MAGENTA",
    "window7="                  "LIGHT-WHITE,BROWN",
    "window8="                  "BLUE,WHITE",

    "window9="                  "LIGHT-CYAN,BLACK",
    "window10="                 "LIGHT-WHITE,BLUE",
    "window11="                 "BLACK,GREEN",
    "window12="                 "BLUE,CYAN",
    "window13="                 "BLUE,RED",
    "window14="                 "LIGHT-WHITE,MAGENTA",
    "window15="                 "WHITE,BROWN",
    "window16="                 "BLACK,WHITE",

    NULL
    };


/*
 *  Default keyboard mapping
 */
const struct k_tbl          x_key_table[] = {
    /* Function keys */
    { F(1),                     "change_window" },
    { F(2),                     "move_edge" },
    { F(3),                     "create_edge" },
    { F(4),                     "delete_edge" },
    { F(5),                     "search_fwd" },
    { F(6),                     "translate" },
    { F(7),                     "remember" },
    { F(8),                     "playback" },
    { F(9),                     "load_macro" },
    { F(10),                    "execute_macro" },
    { SF(7),                    "pause" },

    /* Special functions keys */
    { KEY_HELP,                 "help" },
    { KEY_CUT_CMD,              "cut" },
    { KEY_COPY_CMD,             "copy" },
    { KEY_PASTE,                "paste" },
    { KEY_UNDO_CMD,             "undo" },
    { KEY_REDO,                 "redo" },
    { KEY_EXIT,                 "exit" },
    { KEY_COMMAND,              "execute_macro" },

    /* Keypad */
    { KEY_INS,                  "paste" },
    { KEY_END,                  "end_of_line" },
    { KEY_DOWN,                 "down" },
    { KEY_PAGEDOWN,             "page_down" },
    { WHEEL_DOWN,               "page_down" },
    { KEY_LEFT,                 "left" },
    { KEY_RIGHT,                "right" },
    { KEY_HOME,                 "beginning_of_line" },
    { KEY_UP,                   "up" },
    { KEY_PAGEUP,               "page_up" },
    { WHEEL_UP,                 "page_up" },
    { KEY_DEL,                  "delete_block" },
    { KEY_CUT,                  "cut" },
    { KEY_COPY,                 "copy" },
    { KEY_UNDO,                 "undo" },

    { SHIFT_KEYPAD_2,           "change_window 2" },
    { SHIFT_KEYPAD_4,           "change_window 3" },
    { SHIFT_KEYPAD_6,           "change_window 1" },
    { SHIFT_KEYPAD_8,           "change_window 0" },
    { CTRL_KEYPAD_1,            "end_of_window" },
    { CTRL_KEYPAD_3,            "end_of_buffer" },
    { CTRL_KEYPAD_7,            "top_of_window" },
    { CTRL_KEYPAD_9,            "top_of_buffer" },

    /* Others */
    { CTRL_X,                   "exit" },
    { ALT_0,                    "drop_bookmark 0" },
    { ALT_1,                    "drop_bookmark 1" },
    { ALT_2,                    "drop_bookmark 2" },
    { ALT_3,                    "drop_bookmark 3" },
    { ALT_4,                    "drop_bookmark 4" },
    { ALT_5,                    "drop_bookmark 5" },
    { ALT_6,                    "drop_bookmark 6" },
    { ALT_7,                    "drop_bookmark 7" },
    { ALT_8,                    "drop_bookmark 8" },
    { ALT_9,                    "drop_bookmark 9" },
    { ALT_A,                    "mark 4" },
    { ALT_B,                    "buffer_list" },
    { ALT_C,                    "mark 2" },
    { ALT_D,                    "delete_line" },
    { ALT_E,                    "edit_file" },
    { ALT_F,                    "feature" },
    { ALT_G,                    "goto_line" },
    { ALT_I,                    "insert_mode" },
    { ALT_J,                    "goto_bookmark" },
    { ALT_K,                    "delete_to_eol" },
    { ALT_L,                    "mark 3" },
    { ALT_M,                    "mark" },
    { ALT_O,                    "output_file" },
    { ALT_P,                    "print" },
    { ALT_Q,                    "quote" },
    { ALT_R,                    "read_file" },
    { ALT_S,                    "search_fwd" },
    { ALT_T,                    "translate" },
    { ALT_U,                    "undo" },
    { ALT_V,                    "version" },
    { ALT_W,                    "write_buffer" },
    { ALT_X,                    "exit" },
    { ALT_Z,                    "shell" },

    { 0,                        NULL }
    };

/*end*/
